OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -88,6 +89,12 @@ public override IEnumerable<Target> GetTargets(Actor self)
 		{
 			yield return target;
 		}
+
+		public override TargetLineNode TargetLineNode(Actor self)
+		{
+			var color = NextActivity == null ? Color.Green : NextActivity.TargetLineNode(self).Color;
```

Comment:
Shouldn't each segment be drawn in the colour that the respective activity specifies?